### PR TITLE
Maintain symbolic expressions containing error for path conditions

### PR DESCRIPTION
**Pull request by Himeshi**
New pull request for issue #47

### CODE REVIEW

**Comment by domainexpert** [comment]
@Himeshi I tried the following example that you sent before:
```
#include <klee/klee.h>

int calculate_output(int x, int y)
{
    int z;

    if (x < 5) {
        z = x + y;
    } else {
        z = x * y;
    }
    return z;
}

int main()
{
    int output;
    int input_x, input_y;
    
    klee_make_symbolic(&input_x, sizeof(input_x), "input_x");
    klee_make_symbolic(&input_y, sizeof(input_y), "input_y");

    klee_track_error(&input_x, "input_x_error");
    klee_track_error(&input_y, "input_y_error");
 
    output = calculate_output(input_x, input_y);

    klee_bound_error(output, "output", 1.3);

    return 0;
}
```
But I did not see the difference in the output error expression from before.

**Comment by Himeshi** [comment]
Sorry I forgot to mention that this should be run with --write-kqueries option. In that case the .kquery file will contain the path condition with error.

I moved the path condition with error expression to the .kquery file since that's where the path conditions are written.

File: tools/klee/main.cpp [hunk 1]
```diff
@@ -508,8 +509,20 @@ void KleeHandler::processTestCase(const ExecutionState &state,
     if (errorMessage || WriteKQueries) {
       std::string constraints;
       m_interpreter->getConstraintLog(state, constraints,Interpreter::KQUERY);
+      ConstraintManager constraintsWithError =
+          state.symbolicError->getConstraintsWithError();
       llvm::raw_ostream *f = openTestFile("kquery", id);
       *f << constraints;
+      *f << "\nPath conditions with error:";
```
Comment (domainexpert) [inline tools/klee/main.cpp]: @Himeshi You may want to ensure that this is output only when --precision is specified. Also, perhaps it's best to output to a separate file, maybe a .kquery_precision_error file for easier processing by a script, but I have no strong opinion on this.

**Review by domainexpert — CHANGES_REQUESTED** [review]
@Himeshi Thank you, but just a small issue.

**Review by domainexpert — APPROVED** [review]

**Pull request merged**